packages/babel-generator/src/generators/flow.ts

```diff
@@ -740,7 +740,28 @@ export function VoidTypeAnnotation(this: Printer) {
 }
 
 export function IndexedAccessType(this: Printer, node: t.IndexedAccessType) {
+  const objectNeedsParens =
```

Comment:
Can the parens printing be moved to https://github.com/babel/babel/blob/main/packages/babel-generator/src/node/parentheses.ts?